.github/workflows/ci-pipeline.yml

```diff
@@ -17,6 +17,45 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v2
-
+  
       - name: Run tests
         run: docker-compose -f docker-compose.test.yml up --build
+  deploy:
+    name: Deploy to Docker Hub and Heroku
+    runs-on: ubuntu-latest
+    needs: test
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: cameronmccormack
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build production app
+        run: docker build --target prod --tag cameronmccormack/todo-app:${{ github.sha }} .
+
+      - name: Tag image as latest production image
+        run: docker tag cameronmccormack/todo-app:${{ github.sha }} cameronmccormack/todo-app:latest
+
+      - name: Push image to Docker Hub 
+        run: docker push cameronmccormack/todo-app:latest && docker push cameronmccormack/todo-app:${{ github.sha }}
+
+      - name: Login to Heroku Container Registry
+        env: 
+          HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}

```

Comment:
Again minor, but note that you can specify this as an environment variable at the job level if you prefer, saving you from repeating it on multiple steps; something like:
```
  deploy:
    name: Deploy to Docker Hub and Heroku
    runs-on: ubuntu-latest
    needs: test
    env:
      HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
    steps:
        ......
```